app/src/main/java/com/infomaniak/drive/data/cache/DriveInfosController.kt

```diff
@@ -129,4 +145,37 @@ object DriveInfosController {
 
         return teamList.filter { drive.teams.account.contains(it.id) }
     }
+
+    fun getCurrentDriveCategories(fileCategoriesIds: Array<Int>? = null): List<Category> {
+        if (fileCategoriesIds?.isEmpty() == true) return emptyList()
+
+        val categories = getRealmInstance().use { realm ->
+            val currentDrive = realm.where(Drive::class.java)
+                .equalTo(Drive::id.name, AccountUtils.currentDriveId)
+                .findFirst()
+
+            currentDrive?.categories?.let {
+                val categories = it.where()
+                    .apply { fileCategoriesIds?.let { `in`(Category::id.name, fileCategoriesIds) } }
+                    .findAll()
+
+                realm.copyFromRealm(categories, 0)
+            }
+        } ?: emptyList()
+
+        val sortedList = fileCategoriesIds?.withIndex()?.associate { it.value to it.index }
+            ?.let { orderList -> categories.sortedBy { orderList[it.id] } }
+
+        return fileCategoriesIds?.let { sortedList } ?: categories
+    }
+
+    fun getCategoryRights(): CategoryRights? {
+        return getRealmInstance().use { realm ->
+            val categoryRights = realm.where(Drive::class.java)
+                .equalTo(Drive::id.name, AccountUtils.currentDriveId)
+                .findFirst()
+                ?.categoryRights
+            categoryRights?.let { realm.copyFromRealm(it, 0) }
```

Comment:
Why a val ? I think it's not important here because we understand, just the `let` it's suffisant